proxy/enfuse/files.go

```diff
@@ -0,0 +1,143 @@
+package enfuse
+
+import (
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+	"context"
+	enfuse "github.com/enfabrica/enkit/proxy/enfuse/rpc"
+	"os"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"time"
+)
+
+var (
+	_ fs.NodeRequestLookuper = &FuseDir{}
+	_ fs.Node                = &FuseDir{}
+
+	_ fs.Node         = &FuseFile{}
+	_ fs.HandleReader = &FuseFile{}
+)
+
+func ConvertToDirent(info []*enfuse.FileInfo) []fuse.Dirent {
+	var fdir []fuse.Dirent
+	for _, i := range info {
+		dt := fuse.DT_File
```

Comment:
The body of this loop could be a method on `FileInfo`, no?

Reply:
I'm generally not a fan of `type Something anotherpkg.SomethingElse` and adding in interfaces on it. I'd rather for the function just to accept the type.

Reply:
Ah yup ignore me - I didn't recognize that as a proto type